### PR TITLE
feat(lang): add hare language support

### PR DESCRIPTION
**Pull request by neruyzo**
I want to test out the language [hare](https://harelang.org/) using helix.

### CODE REVIEW

File: runtime/queries/hare/indents.scm [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+[
+  "{"
+  "}"
+] @branch
+
+[
+  (comment)
+] @ignore
```
Comment (the-mikedavis) [inline runtime/queries/hare/indents.scm:14]: the helix indents mechanism doesn't do anything with `@branch` or `@ignore` captures: we could remove these wholesale or comment them out for now (in the future maybe a branch or ignore concept could be useful)

File: runtime/queries/hare/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+(call_expression
+  (postfix_expression) @function)
+
+(function_declaration
+  name: (identifier) @function)
+
+(parameter (name) @variable.parameter)
+
+(field_access_expression
+  selector: (name) @field)
```
Comment (the-mikedavis) [inline runtime/queries/hare/highlights.scm:148]:
```suggestion
  selector: (name) @variable.other.member)
```

Comment (neruyzo) [reply runtime/queries/hare/highlights.scm]: I change it

**Review by the-mikedavis — APPROVED** [review]
looks good, thanks!

File: languages.toml [hunk 3]
```diff
@@ -1249,3 +1249,16 @@ language-server = { command = "vala-language-server" }
...
+scope = "source.hare"
+injection-regex = "hare"
+file-types = ["ha"]
+roots = []
+comment-token = "//"
+indent = { tab-width = 4, unit = "\t" }
+
+[[grammar]]
+name = "hare"
+source = { git = "https://git.sr.ht/~ecmma/tree-sitter-hare", rev = "bc26a6a949f2e0d98b7bfc437d459b250900a165" }
```
Comment (the-mikedavis) [inline languages.toml:1264]: A note for onlookers: this is the first grammar not hosted on GitHub. I don't think we have anything hardcoded to GitHub anywhere though so this shouldn't be a problem.

**Pull request merged**